### ISSUE TITLE
Allow responsive props to be passed to `size` prop for Toggle component

### ISSUE THREAD

**Issue by bencolwell** [palmetto/palmetto-components#491]
I recently worked with the Toggle component where we needed the `size` prop set to `lg` for mobile and `md` for all other media. I ended up wrapping Toggle in two Media components, but it would have been nicer to be able to pass the responsive props for `size`.

(Note the reason I didn't use `useBreakpoint` is because this component is rendered server-side.)

Actual code:
```
          <Box>
            <Media greaterThanOrEqual="tablet">
              <Toggle
                id="include-incentives-desktop"
                label={t('steps.review.purchaseMethod.applyIncentives')}
                isChecked={includeIncentives}
                onChange={setIncentives}
                size="md"
              />
            </Media>
            <Media lessThan="tablet">
              <Toggle
                id="include-incentives-mobile"
                label={t('steps.review.purchaseMethod.applyIncentives')}
                isChecked={includeIncentives}
                onChange={setIncentives}
                size="lg"
              />
            </Media>
          </Box>
```

Desired code:
```
          <Box>
            <Toggle
              id="include-incentives-desktop"
              label={t('steps.review.purchaseMethod.applyIncentives')}
              isChecked={includeIncentives}
              onChange={setIncentives}
              size={{ base: 'lg', tablet: 'md' }}
            />
          </Box>
```

**Comment by nathanyoung** [comment]
Related: https://github.com/palmetto/palmetto-components/issues/437

**Comment by nathanyoung** [comment]
https://trello.com/c/VnQBohjP/687-responsive-size-prop-for-form-inputs

**Comment by nathanyoung** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Responsive size prop for form inputs](https://trello.com/c/VnQBohjP/687-responsive-size-prop-for-form-inputs)

**Comment by juanfabrega** [comment]
appreciate the write up @bencolwell  we will definitely put this in the queue.

**Comment by bencolwell** [comment]
Thanks @juanfabrega and @nathanyoung , definitely appreciate the excellent work you're putting into this library!

**Comment by juanfabrega** [comment]
Hey @bencolwell this has been merged here: https://github.com/palmetto/palmetto-components/pull/559

and is included in release --> `0.86.0`
